statefun-e2e-tests/statefun-smoke-e2e-java/src/main/java/org/apache/flink/statefun/e2e/smoke/CommandInterpreter.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.statefun.e2e.smoke;
+
+import java.time.Duration;
+import java.util.Objects;
+import org.apache.flink.statefun.e2e.smoke.generated.Command;
+import org.apache.flink.statefun.e2e.smoke.generated.Commands;
+import org.apache.flink.statefun.e2e.smoke.generated.SourceCommand;
+import org.apache.flink.statefun.e2e.smoke.generated.VerificationResult;
+import org.apache.flink.statefun.sdk.java.*;
+import org.apache.flink.statefun.sdk.java.message.EgressMessage;
+import org.apache.flink.statefun.sdk.java.message.EgressMessageBuilder;
+import org.apache.flink.statefun.sdk.java.message.Message;
+import org.apache.flink.statefun.sdk.java.message.MessageBuilder;
+import org.apache.flink.statefun.sdk.java.types.SimpleType;
+import org.apache.flink.statefun.sdk.java.types.Type;
+
+public final class CommandInterpreter {
+  private final Ids ids;
+  private static final Duration sendAfterDelay = Duration.ofMillis(1);
+
+  static final Type<Command> COMMAND_TYPE =
+      SimpleType.simpleImmutableTypeFrom(
+          TypeName.typeNameOf("statefun.smoke.e2e.types", "command"),
+          Command::toByteArray,
+          bytes -> Command.parser().parseFrom(bytes));
+
+  static final Type<Commands> COMMANDS_TYPE =
+      SimpleType.simpleImmutableTypeFrom(
+          TypeName.typeNameOf("statefun.smoke.e2e.types", "commands"),
+          Commands::toByteArray,
+          bytes -> Commands.parser().parseFrom(bytes));
+
+  static final Type<SourceCommand> SOURCE_COMMAND_TYPE =
+      SimpleType.simpleImmutableTypeFrom(
+          TypeName.typeNameOf("statefun.smoke.e2e.types", "source-command"),
+          SourceCommand::toByteArray,
+          bytes -> SourceCommand.parser().parseFrom(bytes));
+
+  static final Type<VerificationResult> VERIFICATION_RESULT_TYPE =
+      SimpleType.simpleImmutableTypeFrom(
+          TypeName.typeNameOf("statefun.smoke.e2e.types", "verification-result"),
+          VerificationResult::toByteArray,
+          bytes -> VerificationResult.parser().parseFrom(bytes));
+
+  static final TypeName DISCARD_EGRESS_TYPENAME =
+      TypeName.typeNameOf("statefun.smoke.e2e.types", "discard-egress");
```

Comment:
I believe the egress typenames should be something different. It should match whatever they were defined to be in the driver framework's embedded module bindings.

Reply:
You mean to match the names here?
https://github.com/apache/flink-statefun/blob/dbf62f7e36021897e8010e66a66fdb9e96ce4e3c/statefun-e2e-tests/statefun-smoke-e2e/src/main/java/org/apache/flink/statefun/e2e/smoke/Constants.java

I though we should rename those names in driver framework to be more descriptive names. What do you think?